### ISSUE TITLE
Execute Failing on WalletConnect Approve Token

### ISSUE THREAD

**Issue by earth2travis** [HausDAO/daohaus-monorepo#949]
### Feature

WalletConnect

### App

[Admin](https://admin.daohaus.fun)

### Expected Behavior

Approve CoW Swap to use WETH from the Gnosis Safe

### Actual Behavior

The toast comes up saying it succeeded but...

Transaction is failing when trying to execute ([Transaction Details](https://goerli.etherscan.io/tx/0x89cb69a853651b293ba7031eba88626944f17370d1fe20d1968d2858b131a311))

### Steps to Reproduce

1. Create WalletConnect proposal to approve WETH
2. Vote to pass
3. Wait for grace
4. Try to Execute

### Browser

Brave

### Screenshots and Context

<img width="1010" alt="image" src="https://user-images.githubusercontent.com/522182/194173372-0b1f9fb8-c71a-4660-be7e-e535185bc5d4.png">

<img width="1010" alt="image" src="https://user-images.githubusercontent.com/522182/194173400-a187889a-4784-4195-9df8-942a7165a688.png">

<img width="1010" alt="image" src="https://user-images.githubusercontent.com/522182/194173433-7efa5377-ead2-4471-bdb9-392fbac7e2b9.png">

<img width="1010" alt="image" src="https://user-images.githubusercontent.com/522182/194173510-7263d195-aa62-42ec-9441-168ac4f606d1.png">

<img width="1010" alt="image" src="https://user-images.githubusercontent.com/522182/194173536-a34152c6-f2c9-4620-bac2-c3582b23d370.png">

<img width="1010" alt="image" src="https://user-images.githubusercontent.com/522182/194173566-5dc12595-3fcd-49e8-8b9d-ec33dcd7987e.png">

**Comment by santteegt** [comment]
Cow emojis in the proposal details makes the contract to consume all gas when executing the action. When the proposal was created, the Gnosis API estimated a gaslimit (baalGas) of 107131.  When you call Execute, the app sends baalGas + some extra -> 257131, but this is not enough and it breaks when the contract tries to read the proposal from storage. More specifically:

```
function processProposal(...) {
    ....
    require(state(id) == ProposalState.Ready, "!ready");
    |
    |
    --> function state(uint32 id) public view returns (ProposalState) {
            Proposal memory prop = proposals[id];
            ....
        }
    ....
}
```

Stacktrace:
https://dashboard.tenderly.co/tx/goerli/0x63354b03c58201520a086264bc632dee27c49595446b0211244ae16e56688ead/debugger?trace=0.0.0

**Comment by santteegt** [comment]
~A solution could be to change the contract to read the proposal state variable as storage instead of memory https://github.com/HausDAO/Baal/blob/feat/baalZodiac/contracts/Baal.sol#L902~

**Comment by skuhlmann** [comment]
Can we close this or is more work needed?

@santteegt @earth2travis

**Comment by santteegt** [comment]
need to replicate this after audited contracts are deployed

**Comment by earth2travis** [comment]
@santteegt the contracts, subgraphs and ui have been updated. You should be good to go

**Comment by earth2travis** [comment]
Should probably summon a new DAO

**Comment by santteegt** [comment]
Tested again with the new version of the Baal contract. The exception error found when trying to execute proposals with long proposal descriptions was fixed in #1035
